python/lsst/validate/drp/validate.py

```diff
@@ -0,0 +1,368 @@
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+from lsst.afw.table import SourceCatalog, SchemaMapper, Field
+from lsst.afw.table import MultiMatch, SimpleRecord, GroupView
+import lsst.daf.persistence as dafPersist
+import lsst.pipe.base as pipeBase
+
+from .calcSrd import calcAM1, calcAM2
+from .check import checkAstrometry, checkPhotometry, positionRms
+from .print import printPA1, printPA2, printAM1, printAM2
+from .plot import plotAstrometry, plotPhotometry, plotPA1, plotAM1, plotAM2
+from .util import getCcdKeyName, repoNameToPrefix
+
+
+def loadDataIdsAndParameters(configFile):
+    """Load data IDs, magnitude range, and expected metrics from a yaml file.
+
+    Inputs
+    ------
+    configFile : str
+        YAML file that stores visit, filter, ccd,
+        good_mag_limit, medianAstromscatterRef, medianPhotoscatterRef, matchRef
+
+    Returns
+    -------
+    dict, float, float, float
+        dataIds, good_mag_limit, medianRef, matchRef
+    """
+    stream = open(configFile, mode='r')
+    data = yaml.load(stream)
+
+    ccdKeyName = getCcdKeyName(data)
+    visitDataIds = constructDataIds(data['visits'], data['filter'],
+                                    data[ccdKeyName], ccdKeyName)
+
+    return (visitDataIds,
+            data['good_mag_limit'],
+            data['medianAstromscatterRef'],
+            data['medianPhotoscatterRef'],
+            data['matchRef'],
+           )
+
+
+def constructDataIds(visits, filter, ccds, ccdKeyName=None):
+    """Construct a comprehensive set of Butler Data IDs for a repository.
+
+    Inputs
+    ------
+    visits : list of int
+    filter : str
+    ccds : list of int
+    ccdKeyName : str, optional; default 'ccd'
+        Name to distinguish different parts of a focal plane.
+        Generally 'ccd', but might be 'ccdnum', or 'amp', or 'ccdamp'.
+        Refer to your `obs_*/policy/*Mapper.paf`.
+
+    Returns
+    -------
+    list
+        dataIDs suitable to be used with the LSST Butler.
+
+    Examples
+    --------
+    >>> dataIds = constructDataIds([100, 200], 'r', [10, 11, 12])
+    >>> print(dataIds)
+    [{'filter': 'r', 'visit': 100, None: 10}, {'filter': 'r', 'visit': 100, None: 11}, {'filter': 'r', 'visit': 100, None: 12}, {'filter': 'r', 'visit': 200, None: 10}, {'filter': 'r', 'visit': 200, None: 11}, {'filter': 'r', 'visit': 200, None: 12}]
+
+    Note
+    -----
+    Currently assumes `filter` is a scalar string, e.g., 'g' or 'r-1692 CFHT'.
+    This isn't fundamentally necessary, but one would need to define logic
+    such that filter="ugriz" and filter="r-1692 CFHT" are each processed correctly.
+    """
+    visitDataIds = [{'visit': v, 'filter': filter, ccdKeyName: c}
+                    for v in visits
+                    for c in ccds]
+
+    return visitDataIds
+
+
+def loadRunList(configFile):
+    """Load run list rom a yaml file.
+
+    Inputs
+    ------
+    configFile : str
+        YAML file that stores visit, filter, ccd,
```

Comment:
Probably worth emphasizing that the YAML uses the camera-native names for these variables -- that is, uses `ccdnum` rather than `ccd`. Maybe also point out the format you expect in the YAML (eg a list of CCDs, even if there's only one of them), unless the code is smart enough to correct for this itself (but I don't think it is).